app/services/host_inventory_api.rb

```diff
@@ -7,16 +7,17 @@
 # Interact with the Insights Host Inventory. Usually HTTP calls
 # are all that's needed.
 class HostInventoryAPI
-  def initialize(host, account, url, b64_identity)
-    @host = host
+  def initialize(account, url)
     @url = "#{URI.parse(url)}#{ENV['PATH_PREFIX']}/inventory/v1/hosts"
     @account = account
-    @b64_identity = b64_identity
+    @b64_identity = Base64.strict_encode64(account.to_identity_header.to_json)
   end
 
-  def host_already_in_inventory
-    response = connection.get(@url, {}, 'X_RH_IDENTITY' => @b64_identity)
-    find_results(JSON.parse(response.body))
+  def hosts_already_in_inventory(hosts)
+    response = connection.get(
```

Comment:
The reason why I used 100 is that the inventory is meant to support that number of IDs max - pagination in the UI would not work otherwise with 100 elements

Reply:
I guess what I mean is, this is a public method - perhaps we should ensure `hosts.length < 100`. It's not that important, but I could see someone in the future (mis)using this method by passing in an unbounded hosts list.